providers/inwx/inwxProvider.go

```diff
@@ -0,0 +1,292 @@
+package inwx
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+
+	"github.com/svenpeter42/goinwx"
+)
+
+/*
+INWX Registrar and DNS provider
+
+Info required in `creds.json`:
+	- username
+	- password
+	- totp (TOPT code if 2FA is enabled)
+
+Additional settings available in `creds.json`:
+	- sandbox (set to 1 to use the sandbox API from INWX)
+
+*/
+
+type InwxApi struct {
+	client  *goinwx.Client
+	sandbox bool
+}
+
+var InwxDefaultNs = []string{"ns.inwx.de", "ns2.inwx.de", "ns3.inwx.eu", "ns4.inwx.com", "ns5.inwx.net"}
+var InwxSandboxDefaultNs = []string{"ns.ote.inwx.de", "ns2.ote.inwx.de"}
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Cannot("INWX does not support the ALIAS or ANAME record type."),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseDS:               providers.Unimplemented("DS records require a different API call that hasn't been implemented yet."),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseNAPTR:            providers.Can(),
+	providers.CanUseSRV:              providers.Can("SRV records with empty targets are not supported."),
+	providers.CanUseSSHFP:            providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUseTXTMulti:         providers.Cannot("INWX only supports a single entry for TXT records"),
+	providers.CanAutoDNSSEC:          providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocCreateDomains:       providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAzureAlias:       providers.Cannot(),
+}
+
+func init() {
+	providers.RegisterRegistrarType("INWX", newInwxReg)
+	providers.RegisterDomainServiceProviderType("INWX", newInwxDsp, features)
+}
+
+func newInwx(m map[string]string) (*InwxApi, error) {
+	for key := range m {
+		switch key {
+		case "username",
+			"password",
+			"totp",
+			"sandbox",
+			"domain":
+			continue
+		default:
+			return nil, fmt.Errorf("INWX: unknown key in `creds.json` (%s)\n", key)
+		}
+	}
```

Comment:
I think this should be a warning, not an error.  The reason is that sometimes people include a key like "comment" or "owner".  On the other hand, if someone misspells `passwrod` it would be good to get a warning.

Reply:
good point, fixed.

Reply:
hrm, i've looked at the other provides and only found a similar check in AXFRDDNS: maybe there shouldn't even be a warning. Further down I already make sure that username and password are present and that should be enough to notice if e.g. password was misspelled. 

What are your thoughts on this? I would remove the warning to keep the behavior consistent with (almost) all other providers.

Reply:
On second thought... remove it.  If someone does use (for example) "comment" they'd get a warning in every run and it would be super annoying.